src/webapp/components/form/FieldWidget.tsx

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+
+import { TextInput } from "../text-input/TextInput";
+import { MemberSelector } from "../member-selector/MemberSelector";
+import { MultipleSelector } from "../selector/MultipleSelector";
+import { Selector } from "../selector/Selector";
+import { RadioButtonsGroup } from "../radio-buttons-group/RadioButtonsGroup";
+import { TextArea } from "../text-input/TextArea";
+import { DatePicker } from "../date-picker/DatePicker";
+import { Checkbox } from "../checkbox/Checkbox";
+import { FormFieldState, updateFieldState } from "./FormState";
+
+export type FieldWidgetProps = {
+    onChange: (updatedField: FormFieldState) => void;
+    field: FormFieldState;
+    disabled?: boolean;
+};
+
+export const FieldWidget: React.FC<FieldWidgetProps> = React.memo(props => {
+    const { field, onChange, disabled = false } = props;
+
+    switch (field.type) {
+        case "select": {
+            return field.multiple ? (
+                <MultipleSelector
+                    id={field.id}
+                    placeholder={field.placeholder}
+                    label={field.label}
+                    selected={field.value}
+                    onChange={newValue => onChange(updateFieldState(field, newValue))}
+                    options={field.options}
+                    helperText={field.helperText}
+                    errorText={field.errors ? field.errors.join("/n") : ""}
+                    error={field.errors && field.errors.length > 0}
+                    required={field.required && field.showIsRequired}
+                    disabled={disabled}
+                />
+            ) : (
+                <Selector
+                    id={field.id}
+                    placeholder={field.placeholder}
+                    label={field.label}
+                    selected={field.value}
+                    onChange={newValue => onChange(updateFieldState(field, newValue))}
+                    options={field.options}
+                    helperText={field.helperText}
+                    errorText={field.errors ? field.errors.join("/n") : ""}
+                    error={field.errors && field.errors.length > 0}
+                    required={field.required && field.showIsRequired}
+                    disabled={disabled}
+                />
+            );
+        }
+
+        case "radio": {
+            return (
+                <RadioButtonsGroup
+                    id={field.id}
+                    label={field.label}
+                    selected={field.value}
+                    onChange={event => onChange(updateFieldState(field, event.target.value))}
```

Comment:
All other components return nice values instead of a low-level event, we can do the same for `RatioButtonsGroup`